eduNEXT/tutor-contrib-mfe-extensions
# feat: allow dynamic configuration of the assets path

# Description

> **Note**: This only works in k8s at the moment.

This adds a small script that runs when the MFE container starts. The script replaces a placeholder string (`MFE_EXTENSIONS_PLACEHOLDER_STRING`) with the value of the `MFE_EXTENSIONS_CDN_URL` setting. The placeholder string is used in a custom MFE image and will be prepended to assets path used in the final build.

## Implementation

The script is used as the entrypoint for the MFE container in the pod definition. This guarantees that the script is always run when the container starts. In addition, we explicitly define the initial command (`caddy run ...`) because overriding the entrypoint seems to have wiped out the initial command (`caddy run ...` is the default command used in the Caddy/MFE image and can be checked by using `docker inspect`)

Note that Docker's `entrypoint` corresponds to the `command` argument in the pod definition and Docker's `cmd` correspond to `args`.

The script uses `find` and `sed` to replace all the occurrences of the placeholder string inside the dist folder (the one that holds the output of the webpack build) and runs the initial command.

## How to test

1. Build a new MFE image with the Plugin enabled.
2. Deploy the image **without** defining `MFE_EXTENSIONS_CDN_URL`.
3. Check the network tab while loading any MFE and confirm that the data is being served by the cluster.
4. Define the `MFE_EXTENSIONS_CDN_URL` with the url of a properly configured CDN endpoint and redeploy using the same image.
5. Check the network tab while loading any MFE and confirm that the data is being served by the CDN.
